packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -169,14 +283,187 @@ describe("Sequence.Revertibles with Remote Edits", () => {
+		const id = collection.add(0, 5, IntervalType.Simple, { foo: "one" }).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.changeProperties(id, { foo: "two" });
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+		const int = collection.getIntervalById(id);
+		assert.equal(int?.properties.foo, "one");
+	});
+	it("remote interval property change interacting with reverting an interval add", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple, { foo: "one" }).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("addInterval", (interval, local, op) => {
+			appendLocalAddToRevertibles(interval, revertibles);
+		});
+		collection.add(2, 7, IntervalType.Simple);
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.changeProperties(id, { foo: "two" });
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+		const int = collection.getIntervalById(id);
+		assert.equal(int?.properties.foo, "two");
+	});
+	it("remote interval property change interacting with reverting an interval change", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple, { foo: "one" }).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("changeInterval", (interval, previousInterval, local, op) => {
+			appendLocalChangeToRevertibles(sharedString, interval, previousInterval, revertibles);
+		});
+		collection.change(id, 3, 8);
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.changeProperties(id, { foo: "two" });
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+		const int = collection.getIntervalById(id);
+		assert.equal(int?.properties.foo, "two");
 	});
```

Comment:
Add a case where there are local and remote property changes, with a property that conflicts and one that doesn't.